lib/levelup.js

```diff
@@ -145,62 +135,43 @@ LevelUP.prototype.isClosed = function () {
   return (/^clos|new/).test(this.db.status)
 }
 
-function maybeError (db, callback) {
-  if (!db._isOpening() && !db.isOpen()) {
-    dispatchError(db, new ReadError('Database is not open'), callback)
-    return true
-  }
-}
-
-LevelUP.prototype.get = function (key_, options, callback) {
-  if (key_ === null || key_ === undefined) {
+LevelUP.prototype.get = function (key, options, callback) {
+  if (key === null || key === undefined) {
     throw new ReadError('get() requires a key argument')
   }
 
-  callback = getCallback(options, callback)
-
   var self = this
-  var key
   var promise
 
+  callback = getCallback(options, callback)
+
   if (!callback) [ callback, promise ] = promisify()
 
   if (maybeError(this, callback)) { return promise }
 
-  options = util.getOptions(options)
-  key = this._codec.encodeKey(key_, options)
-
-  options.asBuffer = this._codec.valueAsBuffer(options)
+  options = getOptions(options)
 
   this.db.get(key, options, function (err, value) {
     if (err) {
       if ((/notfound/i).test(err) || err.notFound) {
-        err = new NotFoundError(
-            'Key not found in database [' + key_ + ']', err)
+        err = new NotFoundError('Key not found in database [' + key + ']', err)
       } else {
         err = new ReadError(err)
       }
       return dispatchError(self, err, callback)
     }
-    if (callback) {
-      try {
-        value = self._codec.decodeValue(value, options)
-      } catch (e) {
-        return callback(new EncodingError(e))
-      }
-      callback(null, value)
-    }
+    if (callback) { callback(null, value) }
```

Comment:
we always have a callback now :)

Reply:
Aah right, of course, duh! :)

Reply:
@juliangruber Noticed you missed that for `.open()` and `.close()`, fixing that as well.

Reply:
Done.